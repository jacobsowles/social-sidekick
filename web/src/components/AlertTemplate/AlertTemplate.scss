.alert {
  padding: 10px;
  margin: 70px 10px 0 0;
  border-radius: 3px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 300px;
  box-sizing: border-box;
  box-shadow: 0 1px 2px 0 rgba(60, 64, 67, 0.3), 0 2px 6px 2px rgba(60, 64, 67, 0.15);

  button {
    margin-left: 20px;
    border: none;
    background-color: transparent;
    cursor: pointer;
  }

  span {
    flex: 2;
  }
}

.alert-error {
  background: #f8d7da;
  color: #aa6b70;

  svg {
    stroke: #aa6b70;
  }
}

.alert-info {
  background: #d1ecf1;
  color: #0c5460;

  svg {
    stroke: #0c5460;
  }
}

.alert-success {
  background: #d4edda;
  color: #155724;

  svg {
    stroke: #155724;
  }
}
