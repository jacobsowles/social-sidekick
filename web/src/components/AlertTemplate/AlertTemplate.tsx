import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

import CloseIcon from './CloseIcon';
import ErrorIcon from './ErrorIcon';
import InfoIcon from './InfoIcon';
import SuccessIcon from './SuccessIcon';
import './AlertTemplate.scss';

export interface AlertTemplateProps {
  close?: () => any;
  message?: string;
  options?: any;
}

const AlertTemplate: FunctionComponent<AlertTemplateProps> = ({ close, message, options }) => {
  const classes = classNames(
    'alert',
    { 'alert-info': options.type === 'info' },
    { 'alert-success': options.type === 'success' },
    { 'alert-error': options.type === 'error' }
  );

  return (
    <div className={classes}>
      {options.type === 'info' && <InfoIcon />}
      {options.type === 'success' && <SuccessIcon />}
      {options.type === 'error' && <ErrorIcon />}
      <span>{message}</span>
      <button onClick={close}>
        <CloseIcon />
      </button>
    </div>
  );
};

export default AlertTemplate;
