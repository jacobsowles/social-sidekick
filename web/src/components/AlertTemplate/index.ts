import AlertTemplate from './AlertTemplate';
export default AlertTemplate;
