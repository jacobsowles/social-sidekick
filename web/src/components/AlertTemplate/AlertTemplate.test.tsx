import React from 'react';

import { init, shallow, toJson } from '@tests/component-test-base';
import AlertTemplate, { AlertTemplateProps } from '@components/AlertTemplate';

describe('AlertTemplate', () => {
  let props: AlertTemplateProps;
  init(() => <AlertTemplate {...props} />, 'div');

  beforeEach(() => {
    props = {
      options: {
        type: undefined
      }
    };
  });

  it('should render the component with error icon without crashing', () => {
    props.options.type = 'error';
    toJson(shallow()).should.matchSnapshot();
  });

  it('should render the component with info icon without crashing', () => {
    props.options.type = 'info';
    toJson(shallow()).should.matchSnapshot();
  });

  it('should render the component with success icon without crashing', () => {
    props.options.type = 'success';
    toJson(shallow()).should.matchSnapshot();
  });
});
