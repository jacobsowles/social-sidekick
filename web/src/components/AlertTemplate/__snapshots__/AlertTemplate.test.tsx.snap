// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertTemplate should render the component with error icon without crashing 1`] = `
<div
  className="alert alert-error"
>
  <ErrorIcon />
  <span />
  <button>
    <CloseIcon />
  </button>
</div>
`;

exports[`AlertTemplate should render the component with info icon without crashing 1`] = `
<div
  className="alert alert-info"
>
  <InfoIcon />
  <span />
  <button>
    <CloseIcon />
  </button>
</div>
`;

exports[`AlertTemplate should render the component with success icon without crashing 1`] = `
<div
  className="alert alert-success"
>
  <SuccessIcon />
  <span />
  <button>
    <CloseIcon />
  </button>
</div>
`;
